crates/core/src/ring.rs

```diff
@@ -660,7 +660,11 @@ impl Ring {
             tracing::info!(%peer, "Removing connection");
         }
         self.live_tx_tracker.prune_transactions_from_peer(&peer);
-        let loc = self.location_for_peer.write().remove(&peer).unwrap();
+        let Some(loc) = self.location_for_peer.write().remove(&peer) else {
+            self.open_connections
```

Comment:
maybe add acomment here sayign why we do this:
// this case would be when a connection is being open, so peer location hasn't been recorded yet and we can ignore everything below

Reply:
added